docs/kb/http.md

```diff
@@ -0,0 +1,142 @@
+q)/ request gzipped data, which is unzipped & returned in json and formatted appropriately
+q).j.k .Q.hg "http://httpbin.org/gzip"
+gzipped| 1b
+headers| `Accept-Encoding`Host`X-Amzn-Trace-Id!("gzip";"httpbin.org";"Root=1-665710aa-50bd49d724b532913348a62a")
+method | "GET"
+origin | "78.147.173.108"
+```
+
+In addition, kdb+ provides a low level HTTP request mechanism
+
+```q
+`:http://host:port "string to send as HTTP request"
+```
+
+which returns the HTTP response as a string. 
+
+A HTTP request generally consists of 
+
+* a request line (URL,method,protocol version), terminated by a carriage return and line feed, 
+* zero or more header fields (field name, colon, field value), terminated by a carriage return and line feed
+* an empty line (consisting of a carriage return and a line feed) 
+* an optional message body
+
+e.g.
+
+```q
+q)/ perform HTTP DELETE
+q)`:http://httpbin.org "DELETE /anything HTTP/1.1\r\nConnection: close\r\nHost: httpbin.org\r\n\r\n"
+"HTTP/1.1 200 OK\r\ndate: Wed, 29 May 2024 12:23:54 GMT\r\ncontent-type: application/json\r\ncontent-length: 290\r\nconnection: close\r\nserver: gunicorn/19.9.0\r\naccess-control-allow-origin: *\r\naccess-control-allow-credentials: true\r\n\r\n{\n  \"args\": {},...
+q)postdata:"hello"
+q)/ perform HTTP POST (inc Content-length to denote the payload size)
+q)`:http://httpbin.org "POST /anything HTTP/1.1\r\nConnection: close\r\nHost: httpbin.org\r\nContent-length: ",(string count postdata),"\r\n\r\n",postdata
+"HTTP/1.1 200 OK\r\ndate: Wed, 29 May 2024 13:08:41 GMT\r\ncontent-type: application/json\r\ncontent-length: 321\r\nconnection: close\r\nserver: gunicorn/19.9.0\r\naccess-control-allow-origin: *\r\naccess-control-allow-credentials: true\r\n\r\n{\n  \"args\": {}, \n  \"data\": \"hello\"...
+```
+
+A HTTP response typically consists of
+
+* a status line (protocol version, status code, reason), terminated by a carriage return and line feed
+* zero or more header fields (field name, colon, field value), terminated by a carriage return and line feed
+* an empty line (consisting of a carriage return and a line feed)
+* an optional message body
+
+e.g.
+
+```q
+q)/ x will be complete HTTP response
+q)x:`:http://httpbin.org "DELETE /delete HTTP/1.1\r\nConnection: close\r\nHost: httpbin.org\r\n\r\n"
+q)/ seperate body from headers, get body
+q)@["\r\n\r\n" vs x;1]
+"{\n  \"args\": {}, \n  \"data\": \"\", \n  \"files\": {}, \n  \"form\": {}, \n  \"headers\": {\n    \"Host\": \"httpbin.org\", \n    \"X-Amzn-Trace-Id\": \"Root=1-66572924-7396cee34f268fcd406e94d5\"\n  }, \n  \"json\": null, \n  \"origin\": \"78.147.173.108\", \n  \"url\": \"http://httpbin.org/delete\"\n}\n"
+```
+
+!!! note "If a server uses chunked transfer encoding, the response will be constructed from the chunks prior to returning (since V3.3 2014.07.31)"
```

Comment:
```suggestion
!!! note "If a server uses chunked transfer encoding, the response is constructed from the chunks prior to returning (since V3.3 2014.07.31)."
```